server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java

```diff
@@ -677,14 +689,324 @@ public void testCacheWithInvalidation() throws Exception {
+        // cache should only contain the third entry, the first 2 entries should have been cleaned up
+        assertEquals(expectedThirdCachedEntrySize, currentMemorySize);
+    }
+
+    // cache cleaner wakes up and cleans regularly when staleness threshold is explicitly set to 0
+    public void testStaleKeysCleanup_ZeroStaleThresholdCleansUp() throws Exception {
+        Instant start = Instant.now();
+        long thresholdInMillis = 1_500;
+        String node = internalCluster().startNode(
+            Settings.builder()
+                .put(IndicesRequestCache.SETTING_INDICES_REQUEST_CACHE_STALENESS_THRESHOLD_SETTING, 0)
+                .put(IndicesRequestCache.SETTING_INDICES_REQUEST_CACHE_CLEAN_INTERVAL_SETTING, TimeValue.timeValueMillis(thresholdInMillis))
+        );
+        String index = "index";
+        Client client = client(node);
+        setupIndex(client, index);
+
+        // create first cache entry
+        createCacheEntry(client, index, "hello");
+        assertCacheState(client, index, 0, 1);
+        long expectedFirstCachedItemEntrySize = getRequestCacheStats(client, index).getMemorySizeInBytes();
+        assertTrue(expectedFirstCachedItemEntrySize > 0);
+
+        // create second cache entry
+        createCacheEntry(client, index, "there");
+        assertCacheState(client, "index", 0, 2);
+        assertEquals(expectedFirstCachedItemEntrySize * 2, getRequestCacheStats(client, "index").getMemorySizeInBytes());
+
+        // force refresh so that it creates 2 stale keys in the cache for the cache cleaner to pick up.
+        flushAndRefresh("index");
+        client.prepareIndex("index").setId("1").setSource("k", "good bye");
+        ensureSearchable("index");
+
+        // create another entry
+        createCacheEntry(client, index, "hello1");
+        assertCacheState(client, "index", 0, 3);
+        long currentMemorySize = getRequestCacheStats(client, "index").getMemorySizeInBytes();
+        long expectedThirdCachedEntrySize = currentMemorySize - (expectedFirstCachedItemEntrySize * 2);
+        assertTrue(currentMemorySize > expectedFirstCachedItemEntrySize * 2);
+        assertEquals(currentMemorySize, expectedFirstCachedItemEntrySize * 3, 2);
+
+        Instant end = Instant.now();
+        long elapsedTimeMillis = Duration.between(start, end).toMillis();
+        // if this test is flaky, increase the sleep time.
+        long sleepTime = (thresholdInMillis - elapsedTimeMillis) + 2_500;
+        Thread.sleep(sleepTime);
+
+        // cache cleaner should have cleared the stale keys by now
+        currentMemorySize = getRequestCacheStats(client, "index").getMemorySizeInBytes();
+        // cache should only contain the third entry, the first 2 entries should have been cleaned up
+        assertEquals(expectedThirdCachedEntrySize, currentMemorySize);
+    }
+
+    // cache cleaner wakes up and cleans regularly when staleness threshold is explicitly set to 0
```

Comment:
copy/paste error in comment, setting for stale threshold is not set here. You can refactor this test with above test where 0 is explicitly set.